### ISSUE TITLE
qpdf_set_linearization not working (qpdf 9.0.0)

### ISSUE THREAD

**Issue by markjforte** [qpdf/qpdf#362]
In QPDF 9.0.0 it seems that qpdf_set_linearization is no longer working.  I tried QPDF 8.4.2 and that version seems to work as it should.  

QPDF 9.0.0 produces a PDF, but when I open the PDF in Acrobat it shows Fast Web View: No.  I can also see that the PDF produced by 9.0.0 has a file size a little bit smaller than the one produced by 8.4.2.

FYI, I'm using the prebuilt binaries through the C API.  My code is similar to this example:
```c
    if (((qpdf_read(qpdf, infile, password) & QPDF_ERRORS) == 0) &&
	((qpdf_init_write(qpdf, outfile) & QPDF_ERRORS) == 0))
    {
	qpdf_set_linearization(qpdf, QPDF_TRUE);
	qpdf_write(qpdf);
    }
```

**Comment by jberkenbilt** [comment]
Do you have a file you can share? qpdf has a thorough test suite, and at least in my test suite, the contents of linearized files has not changed except in one rare case of Encryption dictionaries with certain types of errors in them. Note that qpdf 9 is not binary compatible with qpdf 8. If you happen to be using an executable built qpdf 8 and are somehow bypassing the version checks in either shared libraries or DLLs, you have to rebuild the application with qpdf 9. Most likely failing to do that would result in a crash, but it's not inconceivable that doing that with the C API could result in a program that produces incorrect results. The qpdf release is built to avoid that though, so something unusual would have had to happen for that to be the cause.

As a test, I linearized the PDF 1.7 spec with 8.4.0 and 9.0.0 of qpdf. The files were the same size. Adobe Reader DC had "Fast Web View: yes" for both files, and the files were bit-for-bit identical except the /ID field, which should be different every time (unless deterministic ID is specified).

**Comment by markjforte** [comment]
I've tested with a couple PDFs, and seeing the same results.  I've attached one of the smaller test PDFs that I use.

FYI, I'm loading the qpdfXX.dll using the standard Windows API LoadLibrary.  I'm doing so from Delphi.  We've been using QPDF for years in this manner.  When I use the binaries in both qpdf-9.0.0-bin-mingw32.zip and qpdf-9.0.0-bin-msvc32.zip (i.e. qpdf26.dll), neither of them are producing a linearized PDF.  When I use the binaries in qpdf-8.4.2-bin-mingw32.zip (i.e. qpdf21.dll)) I do get a linearized PDF.

[TheDam.pdf](https://github.com/qpdf/qpdf/files/3612158/TheDam.pdf)

Thanks in advance for your assistance.  And thanks for your work on this project!
